### ISSUE TITLE
`prefer-array-some`: Work through variable assignment

### ISSUE THREAD

**Issue by Exodustx0** [sindresorhus/eslint-plugin-unicorn#996]
The `prefer-array-some` rule is not triggered when the `.find()` result is assigned to a variable:
```js
const arr = [];
const hasFoo = arr.find(val => val === 'foo');
if (hasFoo) console.log('foo');
```

Obviously it shouldn't be triggered if `hasFoo` were to be used elsewhere for anything other than element existence checks, but if that's all it's used for, I'd expect it to trigger the rule.

Variable assignment of these element existence checks is useful when the predicate function is long, but also to generally make code read more prose-like.

**Comment by fisker** [comment]
If someone want implement this, I prefer update this function https://github.com/sindresorhus/eslint-plugin-unicorn/blob/dc7f79b0e0a87d7dc009905c9dc76719d9f5a3ff/rules/utils/boolean.js#L29

So, other rules can apply the same logic

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt and feedback in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1662.
